doofinder/includes/multilanguage/class-wpml.php

```diff
@@ -2,39 +2,163 @@
 
 namespace Doofinder\WP\Multilanguage;
 
-class WPML extends Language_Plugin
-{
+use Doofinder\WP\Helpers\Helpers;
+use Doofinder\WP\Log;
+use Doofinder\WP\Settings\Settings;
 
-    /**
-     * @inheritdoc
-     */
-    public function get_languages()
-    {
-        if (!function_exists('icl_get_languages')) {
-            return array();
-        }
-
-        // "wpml_active_languages" filters the list of the
-        // languages enabled (active) for a site.
-        $languages = apply_filters('wpml_active_languages', null, 'orderby=code&order=desc');
-
-        if (empty($languages)) {
-            return array();
-        }
-
-        // Create associative array with lang code / lang name pairs.
-        // For example 'en' => 'English'.
-        $formatted_languages = array();
-        foreach ($languages as $key => $value) {
-            $translated_name = isset($value['translated_name']) ? $value['translated_name'] : '';
-            if(empty($translated_name)){
-                $translated_name = isset($value['display_name']) ? $value['display_name'] : '';
-            }
-            $formatted_languages[$key] = $translated_name;
-        }
+class WPML implements I18n_Handler
+{
 
-        return $formatted_languages;
-    }
+	/**
+	 * A cached list of available languages in the format "get_languages" expects.
+	 *
+	 * @var array
+	 */
+	private $languages;
+
+	/**
+	 * Instance of a class used to log to a file.
+	 *
+	 * @var Log
+	 */
+	private $log;
+
+	/**
+	 * @since 1.0.0
+	 */
+	public function __construct()
+	{
+		global $sitepress;
```

Comment:
Why are we using globally accessible var like this?

Reply:
It is a standard way of getting WPML config